url3426a97ee4_rtreffer_minidns_tgz-pJ8-de_measite_minidns_minidnsrepl_MiniDnsReplJ8/src/de/measite/minidns/dnsserverlookup/AndroidUsingReflection.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2015-2016 the original author or authors
+ *
+ * This software is licensed under the Apache License, Version 2.0,
+ * the GNU Lesser General Public License version 2 or later ("LGPL")
+ * and the WTFPL.
+ * You may choose either license to govern your use of this software only
+ * upon the condition that you accept all of the terms of either
+ * the Apache License 2.0, the LGPL 2.1+ or the WTFPL.
+ */
+package de.measite.minidns.dnsserverlookup;
+
+import de.measite.minidns.util.PlatformDetection;
+
+import java.lang.reflect.Method;
+import java.net.InetAddress;
+import java.util.ArrayList;
+import java.util.logging.Level;
+
+/**
+ * Try to retrieve the list of DNS server by calling SystemProperties.
+ */
+public class AndroidUsingReflection extends AbstractDNSServerLookupMechanism {
+
+    public static final DNSServerLookupMechanism INSTANCE = new AndroidUsingReflection();
+    public static final int PRIORITY = 1000;
+
+    protected AndroidUsingReflection() {
+        super(AndroidUsingReflection.class.getSimpleName(), PRIORITY);
+    }
+
+    @Override
+    public String[] getDnsServerAddresses() {
+        try {
+            Class<?> SystemProperties =
+                    Class.forName("android.os.SystemProperties");
+            Method method = SystemProperties.getMethod("get",
+                    new Class<?>[] { String.class });
+
+            ArrayList<String> servers = new ArrayList<String>(5);
+
+            for (String propKey : new String[] {
+                    "net.dns1", "net.dns2", "net.dns3", "net.dns4"}) {
+
+                String value = (String)method.invoke(null, propKey);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Catch `InvocationTargetException` explicitly when you call `Method.invoke()`. Use `Throwable.getCause()` or `getTargetException()` in the catch handler to access more information about the underlying exception. See the following diffs on GitHub that make this code change: [diff1](https://github.com/EsfingeFramework/gamification/commit/20792047e6c69457cca03e877f2ada578188f32a#diff-6808c12fe9ddaea042db84f95dd5ba78adab99d0cb18bb4d85916a8723530768L21), [diff2](https://github.com/cucumber-attic/cuke4duke/commit/f4821d400cf2590126f1298a8dff9f7a33d45dc5#diff-80adaffc20c0e6e7d2e42312386ffe1ddc10213bf5b8e8919895fd85d666c8deL34)

Learn more: [InvocationTargetException in the Java Platform, Standard Edition 8 API Specification](https://docs.oracle.com/javase/8/docs/api/java/lang/reflect/InvocationTargetException.html)